services/job_runner.go

```diff
@@ -50,7 +52,11 @@ func (rm *jobRunner) Start() error {
 	}
 	rm.done = make(chan struct{})
 	rm.started = true
+
+	rm.demultiplexStarterWg.Add(1)
 	go rm.demultiplexRuns()
+	rm.demultiplexStarterWg.Wait()
```

Comment:
I had to add `demultiplexStarterWg` because there were scenarios in our tests where we started and stopped so fast, the goroutine didn't have time to spin up.

Sequence of events:
- JobRunner Starts
- JobRunner Stops
- JobRunner go demultiplexRuns starts AFTER stop.
- RunChannel Stops
- invariant of RunChannel stopping before JobRunner is violated.

This violation of the invariant led to a `panic`. As is the norm with race conditions, this was a spurious error.